src/Estimators/src/LeggedOdometry.cpp

```diff
@@ -0,0 +1,907 @@
+        }
+        else if (switching == options[2])
+        {
+            m_pimpl->switching = LOSwitching::useExternalUpdate;
+        }
+        else
+        {
+        	std::cerr << printPrefix << "Setting the switching of fixed frame to default, based on latest active contacts." << std::endl;
+        }
+    }
+
+    std::string velComp;
+    if (lohandle->getParameter("vel_computation_method", velComp))
+    {
+        std::vector<std::string> options{"single", "multiAvg", "multLSJvel", "multiLS"};
+        if (velComp == options[0])
+        {
+            m_pimpl->velComp = LOVelComputation::singleFrame;
+        }
+        else if (velComp == options[1])
+        {
+            m_pimpl->velComp = LOVelComputation::multiFrameAverage;
+        }
+        else if (velComp == options[2])
+        {
+            m_pimpl->velComp = LOVelComputation::multiFrameLeastSquareWithJVel;
+        }
+        else
+        {
+            m_pimpl->velComp = LOVelComputation::multiFrameLeastSquare;
+        }
+    }
+
+    if (!lohandle->getParameter("wLin", m_pimpl->wLin))
+    {
+        m_pimpl->wLin = 1.0;
+    }
+
+    if (!lohandle->getParameter("wAng", m_pimpl->m_initialFixedFrame))
+    {
+        m_pimpl->wAng = 0.5;
+    }
+
+    if (!lohandle->getParameter("wJVel", m_pimpl->m_initialFixedFrame))
+    {
+        m_pimpl->wJVel = 10.0;
+    }
+
+    if (!lohandle->getParameter("reg", m_pimpl->m_initialFixedFrame))
+    {
+        m_pimpl->reg = 1e-6;
+    }
+
+    Eigen::Quaterniond refFrame_quat_world = Eigen::Quaterniond(initialWorldOrientationInRefFrame[0], initialWorldOrientationInRefFrame[1],
+                                                                initialWorldOrientationInRefFrame[2], initialWorldOrientationInRefFrame[3]); // here loaded as w x y z
+    refFrame_quat_world.normalize();
+    Eigen::Vector3d refFrame_p_world;
+    refFrame_p_world << initialWorldPositionInRefFrame[0], initialWorldPositionInRefFrame[1], initialWorldPositionInRefFrame[2];
+    m_pimpl->m_refFrame_H_world = manif::SE3d(refFrame_p_world, refFrame_quat_world);
+
+    auto nrJoints = m_modelComp.kinDyn()->getNrOfDegreesOfFreedom();
+    m_pimpl->m_contactJacobian.resize(m_pimpl->m_spatialDim, m_pimpl->m_spatialDim + nrJoints);
+    m_pimpl->m_contactJacobianBase.resize(m_pimpl->m_spatialDim, m_pimpl->m_spatialDim);
+    m_pimpl->m_contactJacobianShape.resize(m_pimpl->m_spatialDim, nrJoints);
+    m_pimpl->m_contactJacobian.setZero();
+    m_pimpl->m_contactJacobianBase.setZero();
+    m_pimpl->m_contactJacobianShape.setZero();
+    return true;
+}
+
+void LeggedOdometry::Impl::updateInternalContactStates(FloatingBaseEstimators::Measurements& meas,
+                                                       FloatingBaseEstimator::ModelComputations& modelComp,
+                                                       std::map<int, EstimatedContact>& contactStates)
+{
+    for (auto& obs : meas.stampedContactsStatus)
```

Comment:
This could be `const auto& ` right?

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/151/commits/914786f7529f478c66665697e516685f9953078b